lib/ae_mdw/stats.ex

```diff
@@ -0,0 +1,128 @@
+defmodule AeMdw.Stats do
+  @moduledoc """
+  Context module for dealing with Stats.
+  """
+
+  alias AeMdw.Blocks
+  alias AeMdw.Collection
+  alias AeMdw.Db.Format
+  alias AeMdw.Db.Model
+  alias AeMdw.Mnesia
+
+  require Model
+
+  @type stat() :: map()
+  @type sum_stat() :: map()
+  @type cursor() :: binary() | nil
+
+  @typep height() :: Blocks.height()
+  @typep direction() :: Mnesia.direction()
+  @typep limit() :: Mnesia.limit()
+  @typep range() :: {:gen, Range.t()} | nil
+
+  @table Model.Stat
+  @sum_table Model.SumStat
+
+  @spec fetch_stats(direction(), range(), cursor(), limit()) :: {[stat()], cursor()}
+  def fetch_stats(direction, range, cursor, limit) do
+    {:ok, {last_gen, -1}} = Mnesia.last_key(AeMdw.Db.Model.Block)
+
+    range_scope = deserialize_scope(range)
+
+    cursor_scope =
+      case deserialize_cursor(cursor) do
+        nil -> nil
+        cursor when direction == :forward -> {cursor, cursor + limit + 1}
+        cursor -> {cursor, cursor - limit - 1}
+      end
+
+    global_scope = if direction == :forward, do: {1, last_gen}, else: {last_gen, 1}
+
+    case intersect_scopes([range_scope, cursor_scope, global_scope], direction) do
```

Comment:
We could have a fixed and explicit precedence rule. Besides making clear what is the criteria applied in the code it helps to get more clear results because 
http://localhost:4000/stats/gen/900-2000?cursor=1100&limit=100
uses the range scope while 
http://localhost:4000/stats/gen/1100-2000?cursor=1100&limit=100
uses the cursor.

Reply:
There's no precedence, it's the intersection. With cursor = 1100 and scope = 900-2000 we get 1100-2000. This way, when you request range 900-2000, on the second page (with limit = 100) you will get cursor = 1000, with the same scope the user requested initially

Reply:
Isn't it better to replace intersection with fixed criteria. It may cause confusion when someone looks at the URL and need to solve conflicts to know what to expect.

An quick alternative would be to generate the next URL only with the cursor . The range would be only for backwards compatibility (without cursor in the first request) and it would avoid the conflict between path and params that defines two ranges in the same request.

Reply:
These endpoints assume the user will never (it's not part of the API) modify, or play around with the `cursor` parameter. We've already had this discussion: the user is exclusively allowed to use the `next` URL, whatever it may contain. If we decide tomorrow to name the `cursor` parameter "foo", we should be able to do so without any breaking functionality.

Reply:
We can also base64 the cursor parameter to make it more evident that it's not something modifiable.

Reply:
yeah maybe we should make the cursor not so presentable for devs to abuse it

Reply:
but lets discuss this in a separate issue https://github.com/aeternity/ae_mdw/issues/386

Reply:
@sborrazas The point here is not about editing but a simple change that would remove conflicting info. The fact that the dev should just program the request to the `next` it as is doesn't imply that we can encode it in any way (for the sake of argument). Added as *suggestion* in the #386.